golem-component-service-base/src/service/transformer_plugin_caller.rs

```diff
@@ -0,0 +1,168 @@
+// Copyright 2024-2025 Golem Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::model::Component;
+use async_trait::async_trait;
+use golem_common::model::component::ComponentOwner;
+use golem_common::model::RetryConfig;
+use golem_common::retries::with_retries;
+use golem_common::SafeDisplay;
+use http::StatusCode;
+use reqwest::multipart::{Form, Part};
+use std::collections::HashMap;
+use std::fmt::{Debug, Display, Formatter};
+
+#[async_trait]
+pub trait TransformerPluginCaller<Owner: ComponentOwner>: Debug + Send + Sync {
+    async fn call_remote_transformer_plugin(
+        &self,
+        component: &Component<Owner>,
+        data: &[u8],
+        url: String,
+        parameters: &HashMap<String, String>,
+    ) -> Result<Vec<u8>, TransformationFailedReason>;
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum TransformationFailedReason {
+    Failure(String),
+    Request(reqwest::Error),
+    HttpStatus(StatusCode),
+}
+
+impl Display for TransformationFailedReason {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            TransformationFailedReason::Failure(message) => write!(f, "{message}"),
+            TransformationFailedReason::Request(error) => write!(f, "Request error: {error}"),
+            TransformationFailedReason::HttpStatus(status) => write!(f, "HTTP status: {status}"),
+        }
+    }
+}
+
+impl SafeDisplay for TransformationFailedReason {
+    fn to_safe_string(&self) -> String {
+        self.to_string()
+    }
+}
+
+#[derive(Debug)]
+pub struct TransformerPluginCallerDefault {
+    client: reqwest::Client,
```

Comment:
The client maintains an internal connection pool per host. Not recreating it per plugin call should resolve this todo:
https://github.com/golemcloud/golem/pull/1405/files#diff-f4ae9ef1c03598ea0766776182ed99fa41f5b79f88a02fcfd50e4e804f03f3c0L924